pkg/controller/batch/computeenvironment/setup.go

```diff
@@ -0,0 +1,460 @@
+		ComputeEnvironment: awsclients.String(meta.GetExternalName(cr)),
+		State:              awsclients.String(svcsdk.CEStateDisabled)}); err != nil {
+		return true, awsclients.Wrap(err, "UpdateComputeEnvironment-call for CE deletion failed")
+	} //depending on the CE Setup (type, compute resources, status ...) Update-Call can take some while (-> status UPDATING) before getting to the Deletion...
+
+	return true, nil //risky? (ASK)
+	//maybe end here with return true, nil bc "Deletion queued. Disabling CE first."
+	//then postDelete schould be unnecessary?
+	//seems risky to only have one return false that leads to a Deletion request ...
+	//on the other hand it resides behind an if that checks for the DISABLED-State and without that the deletion will not work...
+
+	//endless loop: goes into UPDATING status, after awhile gets VALID status, ...
+	//but state gets disabled and stays that way -> so problem is probably the updating of the computeResources? that enforces the UPDATING Status
+	//get STATE to AtProvider to check for it, so no further updatecalls
+}
+
+func postDelete(_ context.Context, cr *svcapitypes.ComputeEnvironment, obj *svcsdk.DeleteComputeEnvironmentOutput, err error) error {
+
+	//CB: possibly not needed when predelete solves all...
+	if err != nil {
+		if strings.Contains(err.Error(), "modified") {
+			// skip: Can't delete workspace in non-ACTIVE state. Current status is DELETING
+
+			fmt.Println("postDelete skipped err modified ", err)
+			return nil
+		}
+		return err
+	}
+	return err
+}
+
+func isUpToDate(cr *svcapitypes.ComputeEnvironment, obj *svcsdk.DescribeComputeEnvironmentsOutput) (bool, error) {
+
+	fmt.Println("isUpToDate: cr.Status.AtProvider.Status :", awsclients.StringValue(cr.Status.AtProvider.Status))
+	//Skip when updating, deleting or creating
+	if awsclients.StringValue(cr.Status.AtProvider.Status) == string(svcapitypes.CEStatus_UPDATING) ||
+		awsclients.StringValue(cr.Status.AtProvider.Status) == string(svcapitypes.CEStatus_DELETING) ||
+		awsclients.StringValue(cr.Status.AtProvider.Status) == string(svcapitypes.CEStatus_CREATING) {
+		return true, nil
+	}
+
+	ce := obj.ComputeEnvironments[0]
+
+	fmt.Printf("isUpToDate: spec.State: %v\n", awsclients.StringValue(cr.Status.AtProvider.State))
+	if !cmp.Equal(cr.Spec.ForProvider.DesiredState, ce.State) {
+
+		fmt.Println("isUpToDate states false 1/2: cr.Spec.ForProvider.DesiredState ", awsclients.StringValue(cr.Spec.ForProvider.DesiredState))
+		fmt.Println("isUpToDate states false 2/2: ce.State ", awsclients.StringValue(ce.State))
+		return false, nil
+	} //should be unnecessary if below - comparing all fields in one go - works
+	//however checking only relevant fields one by one would be easier in this case (or if other MANAGED type require custom handlings)?
+
+	objspecs := GenerateComputeEnvironment(obj)
+	if err := lateInitialize(&objspecs.Spec.ForProvider, obj); err != nil {
```

Comment:
`LateInitialize` is called by the generated controller (if you add it to the hooks). There should be no need to add it manually.